content/blog/emacs.md

```diff
@@ -0,0 +1,57 @@
+date: 2020-04-01T10:00:00Z
+author: "[Konrad Hinsen](https://khinsen.net/)"
+authorAvatar: khinsen
+tags:
+  - Tool
+featured_image:
+description:
+summary:
+  "Emacs is probably the most successful malleable system in the history
+  of computing. What can we learn from it for designing the malleable
+  systems of the future?"
+---
+
+Malleable systems, as defined in the [mission statement][mission], were more fashionable in the past than they are now. There was a time, in the 1970s and 1980s, when malleability was a goal for influential systems designers. It was most prominently advertised by the Smalltalk design team, but it was strong as well in the Lisp community. [Lisp machines][lisp-machine], for example, were malleable systems. But Lisp machines are gone, and Smalltalk has remained a niche product to this day. Only one malleable system of the 1970s is still in wide use: [Emacs][emacs]. However, "wide use" is relative: it's true only in comparison to other malleable systems. On an absolute scale of software popularity, malleable systems are hardly visible. So what can we learn from the relative success of Emacs for the design of future malleable systems?
+
+## Lesson #1: Provide significant out-of-the-box utility
+
+The demise of Lisp machines can be easily explained by their price tag. So let's compare Emacs to Smalltalk. What did Emacs get right that Smalltalk didn't? I believe that the main aspect was an out-of-the-box utility for a large number of computer users. Emacs is advertised and used as a text editor. It's of course advertised as an *extensible* text editor (see the [GNU Emacs home page][gnu-emacs]), and that claim is easy to verify by a Web search, so it's probably a frequent reason why people choose Emacs over other text editors. But most Emacs users start out as plain users of Emacs' text editing features.
+
+Compare this to a typical Smalltalk system, such as [Pharo][pharo], which I [started to explore][pharo-blog-post] about a year ago. As a malleable environment, Pharo is in many ways superior to Emacs. For example, it has support for malleable graphical user interfaces, where Emacs is limited to text plus embedded inert images. But Pharo on its own isn't good for anything. You have to write code yourself to make it do anything useful for you. In other words, you have to approach Pharo with the mindset of a software developer. Pharo code is certainly easy to change (principle number one of the mission statement), but there is nothing in Pharo that a typical computer end user would *want* to use, and then change. And yet, Pharo has already made a lot of progress compared to earlier Smalltalk systems, which were completely disconnected from mainstream computing. You *can* implement immediately useful tools in Pharo, because you can read files, open HTTP connections, and call non-Pharo code. Someone *could* write a killer app in Pharo that would give it the same attractiveness as Emacs. But it hasn't happened yet.
```

Comment:
For "easy to change", you could link to that bit of the mission if you want, with something like the following:

```
[mission-1]: {{< relref "/pages/mission.md#1-easy-to-change" >}}
```

See the home page source for more examples:

https://github.com/malleable-systems/malleable.systems/blob/f506b70a9988a6693b6a92c72fdbfee491554c3b/content/_index.md#L51-L57

Reply:
Good to know, that's indeed what I would have done had I known how to do it.

Reply:
Done.